src/genkey/clu_genkey.c

```diff
@@ -1274,6 +1274,247 @@ int wolfCLU_genKey_Dilithium(WC_RNG* rng, char* fName, int directive, int fmt,
 #endif /* HAVE_DILITHIUM */
 }
 
+int wolfCLU_genKey_ML_DSA(WC_RNG* rng, char* fName, int directive, int fmt,
+    int keySz, int level, int withAlg)
+{
+#ifdef HAVE_DILITHIUM
+    int    ret = WOLFCLU_SUCCESS;
+
+    XFILE  file       = NULL;
+    int   fNameSz     = 0;
+    int   fExtSz      = 6;  // size of ".priv\0" or ".pub\0\0"
+    char  fExtPriv[6] = ".priv\0";
+    char  fExtPub[6]  = ".pub\0\0";
+    char* fOutNameBuf = NULL;
+
+#ifdef NO_AES
+    /* use 16 bytes for AES block size */
+    size_t maxDerBufSz = 4 * keySz * 16;
+#else
+    size_t maxDerBufSz = 4 * keySz * AES_BLOCK_SIZE;
+#endif  /* NO_AES */
+
+    byte*  derBuf      = NULL;
+    byte*  pemBuf      = NULL;
+    byte*  outBuf      = NULL;
+    int    derBufSz    = -1;
+    int    pemBufSz    = 0;
+    int    outBufSz    = 0;
+
+#ifdef WOLFSSL_SMALL_STACK
+    MlDsaKey* key;
+    key = (MlDsaKey*)XMALLOC(sizeof(MlDsaKey), HEAP_HINT,
+            DYNAMIC_TYPE_DILITHIUM);
+    if (key == NULL) {
+        return MEMORY_E;
+    }
+#else
+    MlDsaKey key[1];
+#endif
+
+    if (rng == NULL || fName == NULL) {
+        return BAD_FUNC_ARG;
```

Comment:
In the small stack case this is going to leak the MlDsaKey pointer from earlier.